src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/StorageManager.java

```diff
@@ -1182,47 +1184,78 @@ private void updateDatabase() {
      * Updates the database to the latest version
      *
      */
-
 	private void updatePlayersToUUID() {
-		plugin.getLogger().log(Level.WARNING, "Starting Migration to UUID Players !");
-		plugin.getLogger().log(Level.WARNING, "==================== ATTENTION DONT STOP BUKKIT ! ==================== ");
-		plugin.getLogger().log(Level.WARNING, "==================== ATTENTION DONT STOP BUKKIT ! ==================== ");
-		plugin.getLogger().log(Level.WARNING, "==================== ATTENTION DONT STOP BUKKIT ! ==================== ");
-        SimpleClans.getInstance().setUUID(false);
+        logMigrationStart();
+
         List<ClanPlayer> cps = retrieveClanPlayers();
+        Map<String, UUID> uuidMap = fetchUUIDs(cps);
 
-        int i = 1;
-        for (ClanPlayer cp : cps) {
+        int totalPlayers = cps.size();
+        for (int i = 0; i < totalPlayers; i++) {
+            ClanPlayer cp = cps.get(i);
             try {
-                UUID uuidPlayer;
-                if (SimpleClans.getInstance().getServer().getOnlineMode()) {
-                    uuidPlayer = UUIDFetcher.getUUIDOfThrottled(cp.getName());
-                } else {
-                    uuidPlayer = UUID.nameUUIDFromBytes(("OfflinePlayer:" + cp.getName()).getBytes(Charsets.UTF_8));
-                }
-                String query = "UPDATE `" + getPrefixedTable("players") + "` SET uuid = '" + uuidPlayer.toString() + "' WHERE name = '" + cp.getName() + "';";
-                core.executeUpdate(query);
+                UUID uuid = uuidMap.getOrDefault(cp.getName(), cp.getUniqueId());
```

Comment:
retrieveClanPlayers() (line 1191) doesn't initialize the UUIDs. This can be null.